From 0aafe286dc9f0ec6100adaaddc50305d60d7fff6 Mon Sep 17 00:00:00 2001
From: Jaydeep Patel <jaydeepkumar.patel@amd.com>
Date: Mon, 6 Nov 2023 11:59:29 +0000
Subject: [PATCH 5/5] SWDEV-430086 - Need to set BUILD_SHARED_LIBS so that it
 gets reflected to rocclr.

Build process was top down Pre CLR (23.10) vs bottom up since
CLR (>=23.20) and so BUILD_SHARED_LIBS value is not being reflected
in rocclr build process since CLR. With this change, BUILD_SHARED_LIBS
is set pre rocclr compilation.

Change-Id: Ia2cd3b8148e9df2df222c1e734d927f2c029017e
---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4e4ad5407..5f30d3151 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -67,7 +67,9 @@ if(CLR_BUILD_HIP)
         endif()
     endif()
 endif()
-
+if(CLR_BUILD_HIP)
+    option(BUILD_SHARED_LIBS "Build the shared library" ON)
+endif()
 if((CLR_BUILD_HIP AND HIP_PLATFORM STREQUAL "amd") OR CLR_BUILD_OCL)
     add_subdirectory(rocclr)
 elseif(HIP_PLATFORM STREQUAL "amd")
-- 
2.43.1

